Content.Server/GameObjects/Components/Weapon/Melee/MeleeChemicalInjectorComponent.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Content.Server.GameObjects.Components.Body.Circulatory;
+using Content.Server.GameObjects.Components.Chemistry;
+using Content.Shared.Chemistry;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using System;
+using System.Collections.Generic;
+
+namespace Content.Server.GameObjects.Components.Weapon.Melee
+{
+    [RegisterComponent]
+    public class MeleeChemicalInjectorComponent : Component
+    {
+        public override string Name => "MeleeChemicalInjector";
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public ReagentUnit TransferAmount { get; set; }
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float TransferEfficiency { get => _transferEfficiency; set => _transferEfficiency = Math.Clamp(value, 0, 1); }
+        private float _transferEfficiency;
+
+        [ViewVariables]
+        private SolutionContainerComponent _solutionContainer;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(this, x => x.TransferAmount, "transferAmount", ReagentUnit.New(1));
+            serializer.DataField(ref _transferEfficiency, "transferEfficiency", 1f);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            var meleeWeapon = Owner.EnsureComponent<MeleeWeaponComponent>();
+            Owner.EntityManager.EventBus.SubscribeEvent<MeleeHitEvent>(EventSource.Local, this, OnMeleeHit);
```

Comment:
This will need to be subscribed to by a system that handles the functionality OR use component messages (from a component you can just call `SendMessage` to use it locally). The problem with eventbus subscriptions on components is that evveerrryyy component will get this message now even if not relevant to its entity.

Reply:
Switched to using component messages